### PR TITLE
Rework class

### PR DESCRIPTION
**Pull request by encukou**
This upsates the lesson on classes and inheritance.
The changes aren't as drastic as in some of my recent PRs :)

I did remove the Cat exercise; IMO it's better to give this as an assignment.

### CODE REVIEW

File: lessons/beginners/class/index.md [hunk 1]
```diff
@@ -149,7 +153,7 @@ aby se nepletly s „normálními“ hodnotami.
 
 V těle třídy můžeš definovat metody, které vypadají
 úplně jako funkce – jen mají první argument `self`.
```
Comment (adelpopelkova) [inline lessons/beginners/class/index.md:155]: I would like to ask about `self`. Here it is an *argument* and later in the materials it is a *parametr*. I am still not 100% sure about the difference between these two terms. Or are the rules for when to use which one of them different for functions and methods?

Comment (encukou) [reply lessons/beginners/class/index.md:155]: They have precise meanings, bit I (and other programmers) unfortunately often mix them up.
It's like the words *zásuvka*/*zástrčka*, both in what they mean and in how people mix them up :)

* A *parameter* is seen in the function definition. It is the variable *name*.
* An *argument* is the *value* passed to the function when the function is called.

```suggestion
úplně jako funkce – jen mají první parametr `self`.
```

File: lessons/beginners/class/index.md [hunk 2]
```diff
@@ -320,60 +340,18 @@ class Kotatko:
...
-        if not self.je_ziva():
-            print("Je zbytecne krmit mrtvou kocku!")
-            return
-        if jidlo == "ryba" and self.pocet_zivotu < 9:
-            self.pocet_zivotu += 1
-            print("Kocka spapala rybu a obnovil se ji jeden zivot.")
-        else:
-            print("Kocka se krmi.")
-```
-{% endfilter %}
```
Comment (encukou) [inline lessons/beginners/class/index.md:376]: @frenzymadness Zvládnete to bez úkolu v materiálech?

Comment (frenzymadness) [reply lessons/beginners/class/index.md:376]: Myslím si, že ano. Navrhnu jeho transformaci na domácí projekt. Máme v plánu začít s domácími projekty ještě na hodině, takže to snad nebude žádná velká změna. Cc @adelpopelkova @Disi77

**Pull request merged**